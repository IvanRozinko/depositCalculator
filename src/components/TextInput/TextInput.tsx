import React from 'react'
import { OutlinedInput, makeStyles } from '@material-ui/core';

const useStyle = makeStyles({
  input: {
    width: 100,
    fontSize: 18
  }
})
interface Props {
  sum: number,
  handleInputChange: (event: React.ChangeEvent<HTMLInputElement>) => void
}
function TextInput({ sum, handleInputChange }: Props ) {
  const styles = useStyle();
  return (
    <OutlinedInput 
      className={styles.input}
      value={sum}
      fullWidth
      onChange={handleInputChange}
    />
  )
}

export default TextInput
