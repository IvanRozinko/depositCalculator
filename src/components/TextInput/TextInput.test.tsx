import TextInput from './TextInput';
import { shallow } from 'enzyme';

describe('<TextInput /> :', () => {
  const sum = 100;
  const handleInputChange = jest.fn();
  
  it('renders correctly', () => {
    const component = shallow(
      <TextInput 
        sum={sum} 
        handleInputChange={handleInputChange} 
      />
    );
    expect(component).toMatchSnapshot();
  })

  it('sum prop be equal to passed prop', () => {
    const component = shallow(
      <TextInput 
        sum={sum} 
        handleInputChange={handleInputChange} 
      />
    );
    expect(component.props().value).toBe(sum);
  })

  it('handleInputChange should be called', () => {
    const component = shallow(
      <TextInput 
        sum={sum} 
        handleInputChange={handleInputChange} 
      />
    );
    component.simulate('change', { target: { value: 'test' } })
    expect(handleInputChange).toBeCalled();
  })
})