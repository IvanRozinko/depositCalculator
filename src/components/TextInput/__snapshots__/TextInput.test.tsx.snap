// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> : renders correctly 1`] = `
<WithStyles(ForwardRef(OutlinedInput))
  className="makeStyles-input-1"
  fullWidth={true}
  onChange={[MockFunction]}
  value={100}
/>
`;

exports[`<TextInput /> renders correctly 1`] = `
<WithStyles(ForwardRef(OutlinedInput))
  className="makeStyles-input-1"
  fullWidth={true}
  onChange={[MockFunction]}
  value={100}
/>
`;

exports[`<TextInput renders correctly 1`] = `
<WithStyles(ForwardRef(OutlinedInput))
  className="makeStyles-input-1"
  fullWidth={true}
  onChange={[MockFunction]}
  value={100}
/>
`;

exports[`TextInput renders correctly 1`] = `
<WithStyles(ForwardRef(OutlinedInput))
  className="makeStyles-input-1"
  fullWidth={true}
  onChange={[MockFunction]}
  value={100}
/>
`;
